firmware-support/bittide-sys/src/time/self_test.rs

```diff
@@ -0,0 +1,150 @@
+use crate::time::Clock;
+use crate::time::Duration;
+use crate::time::Instant;
+
+/// Tests for the time module.
+/// It receives a pointer to the timing peripheral and returns a list of tuples containing the
+/// name of the test and a boolean indicating if the test passed.
+pub fn self_test(clock: Clock) -> [(&'static str, bool); 13] {
```

Comment:
I'm not 100% sure how I would do those tests. They can't be run with the usual testing framework, but they aren't really part of the API either, so I'm not sure what the best place would be. Not really actionable, just something I think I'd like to think about a bit more.

Also like with the `Clock::self_test()` function I would recommend a `-> impl Iterator<Item = (&'static str, bool)>` return instead

Reply:
In that case I think we should just keep the testing module completely separate.
We can include it whenever we deem it necessary and remove it when don't.

I'll take a look at the return types.